### ISSUE TITLE
Edge emitted before vertex is emitted

### ISSUE THREAD

**Issue by jepetty** [Microsoft/lsif-typescript#27]
Running `node .\node_modules\tsc-lsif\lib\main.js -p .\path\to\tsconfig.json --outputFormat=line | node .\node_modules\npm-lsif\lib\main.js .\path\to\package.json` emits an edge before a vertex. I've reproduced this on this [repo](https://github.com/karigari/vscode-chat). In this case, an edge referring to vertex "4868" is emitted before the vertex. I've attached the out file for completeness.
[out1.txt](https://github.com/Microsoft/lsif-typescript/files/2968133/out1.txt)

**Comment by jepetty** [comment]
@arjun27 is unable to reproduce, but is using a mac. Is it possible a problem like this could exist only for Windows users?

**Comment by arjun27** [comment]
Thanks @jepetty. While I was not able to find this issue on vertex id 4868 on the lsif.json generated on my system, I did find other vertex ids that were referenced in edges, before the vertex itself was declared. I found 63 such cases using [this script that prints them out](https://gist.github.com/arjun27/96517f6c334192c8cdf43fd0ea8de895). Incidentally, running the script on the out1.txt attached in @jepetty's message also results in 63 such cases.

I also verified that this issue does not show up in the [vsls-contrib/pomodoro](https://github.com/vsls-contrib/pomodoro) or [vsls-contrib/whiteboard](https://github.com/vsls-contrib/whiteboard) repos.
